data-prepper-plugins/aggregate-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/aggregate/actions/CountAggregateAction.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * C
+ * SPDX-Limense-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.processor.aggregate.actions;
+
+import org.opensearch.dataprepper.model.annotations.DataPrepperPlugin;
+import org.opensearch.dataprepper.model.event.Event;
+import org.opensearch.dataprepper.model.event.JacksonEvent;
+import org.opensearch.dataprepper.plugins.processor.aggregate.AggregateAction;
+import org.opensearch.dataprepper.plugins.processor.aggregate.AggregateActionInput;
+import org.opensearch.dataprepper.plugins.processor.aggregate.AggregateActionResponse;
+import org.opensearch.dataprepper.plugins.processor.aggregate.GroupState;
+
+import java.util.Optional;
+
+/**
+ * An AggregateAction that combines multiple Events into a single Event. This action will add the unique keys of each smaller Event to the overall groupState,
+ * and will create a combined Event from the groupState on concludeGroup. If smaller Events have the same keys, then these keys will be overwritten with the keys of the
+ * most recently handled Event.
+ * @since 1.3
+ */
+@DataPrepperPlugin(name = "count", pluginType = AggregateAction.class)
+public class CountAggregateAction implements AggregateAction {
+    static final String EVENT_TYPE = "event";
+    // COUNTKEY string is chosen to be unique and NOT to conflict with user provided keys
+    // This COUNTKEY is used by OTEL/Prometheus format converter processor to generate a SUMMARY metric
+    public static final String COUNTKEY = "aggr._count";
+
+    @Override
+    public AggregateActionResponse handleEvent(final Event event, final AggregateActionInput aggregateActionInput) {
+        final GroupState groupState = aggregateActionInput.getGroupState();
+        if (groupState.get(COUNTKEY) == null) {
+            groupState.put(COUNTKEY, 1);
```

Comment:
Instead of having the function `useIdenfiticationKeysOnly` and initializing the group state with the identification key map on construction of the aggregate group, can we just pass the `identification_keys` as part of the `AggregateActionInput`, and add those identification_keys and event values to the group state a single time in this branch?

Reply:
That's definitely an option. I have implemented that way first but I thought this is easier/simpler than that approach. When the group is created, we have both keys and values. And the keys and values are sent to group state single time in this implementation too, right?

Reply:
That's right. The identification_keys, along with the countKey of 1, would be added a single time the first time an Event triggers the creation of an AggregateGroup in the `handleEvent` function. So the result would be the same as what you have implemented, but I think that keeping this functionality specific to the AggregateAction logic is simpler than forcing aggregate action developers to understand and implement the the new method, and it could also simply be the case that the groupState always starts as empty. So an empty GroupState can consistently indicate that this is likely the first event for that group (depending on the aggregate action implementation of course)